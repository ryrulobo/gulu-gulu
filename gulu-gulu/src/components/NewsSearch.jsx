import moment from "moment";

export default function NewsSearch({
  results,
  isLogin,
  bookmarkData,
  setBookmarkData,
}) {
  return (
    <div
      className="flex flex-wrap justify-center space-y-6 sm:px-56 p-5"
      data-testid="newsSearch"
    >
      {results.map(
        ({ datePublished, provider, image, name, description, url }, index) => (
          <div className="grid grid-flow-col gap-5" key={index}>
            <div className="w-full">
              <a href={url} target="_blank" rel="noreferrer">
                {provider && provider[0] && (
                  <p className="text-sm">{provider[0].name}</p>
                )}
                <p className="text-lg hover:underline text-blue-700">{name}</p>
              </a>
              <p>{description}</p>
              <p className="text-sm mt-1">
                {moment(datePublished).startOf("day").fromNow()}
              </p>
            </div>
            <div className="flex justify-center items-center">
              {image && image.thumbnail && (
                <img
                  src={image.thumbnail.contentUrl}
                  alt={image.thumbnail.contentUrl}
                  loading="lazy"
                  className="w-30 rounded"
                />
              )}
            </div>
            <div className="flex justify-center items-center pl-1">
              {isLogin ? (
                <button
                  onClick={() => {
                    setBookmarkData({
                      ...bookmarkData,
                      datePublished,
                      provider: provider[0].name,
                      image: !image ? "" : image.thumbnail.contentUrl,
                      name,
                      description,
                      url,
                    });
                  }}
                >
                  <svg
                    id="icon"
                    className="w-6 h-6 hover:fill-blue-500 stroke-blue-500"
                    fill="none"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      d="M5 5a2 2 0 012-2h10a2 2 0 012 2v16l-7-3.5L5 21V5z"
                    ></path>
                  </svg>
                </button>
              ) : null}
            </div>
          </div>
        )
      )}
    </div>
  );
}
