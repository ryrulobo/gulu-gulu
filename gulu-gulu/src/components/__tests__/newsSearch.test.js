import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import NewsSearch from "../NewsSearch";

afterEach(() => {
  cleanup();
});

const results = [
  {
    datePublished: "2022-10-27T19:04:14.922Z",
    provider: "test.com",
    image: { thumbnail: { contentUrl: "test.com/image-thumbnail" } },
    name: "news title",
    description: "this is test title",
    url: "test.com/news-test",
  },
];

const isLogin = true;

test("Render news search", () => {
  render(<NewsSearch results={results} isLogin={isLogin} />);
  const newsSearchElement = screen.getByTestId("newsSearch");
  expect(newsSearchElement).toBeInTheDocument();
  expect(newsSearchElement).toContainHTML("a");
  expect(newsSearchElement).toContainHTML("img");
  expect(newsSearchElement).toContainHTML("p");
  expect(newsSearchElement).toContainHTML("button");
  expect(newsSearchElement).toContainHTML("svg");
  expect(newsSearchElement).toContainHTML("path");
});

test("Matches snapshots", () => {
  const tree = renderer
    .create(<NewsSearch results={results} isLogin={isLogin} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
