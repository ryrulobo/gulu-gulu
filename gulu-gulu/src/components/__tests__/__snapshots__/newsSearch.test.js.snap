// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1`] = `
<div
  className="flex flex-wrap justify-center space-y-6 sm:px-56 p-5"
  data-testid="newsSearch"
>
  <div
    className="grid grid-flow-col gap-5"
  >
    <div
      className="w-full"
    >
      <a
        href="test.com/news-test"
        rel="noreferrer"
        target="_blank"
      >
        <p
          className="text-sm"
        />
        <p
          className="text-lg hover:underline text-blue-700"
        >
          news title
        </p>
      </a>
      <p>
        this is test title
      </p>
      <p
        className="text-sm mt-1"
      >
        15 hours ago
      </p>
    </div>
    <div
      className="flex justify-center items-center"
    >
      <img
        alt="test.com/image-thumbnail"
        className="w-30 rounded"
        loading="lazy"
        src="test.com/image-thumbnail"
      />
    </div>
    <div
      className="flex justify-center items-center pl-1"
    >
      <button
        onClick={[Function]}
      >
        <svg
          className="w-6 h-6 hover:fill-blue-500 stroke-blue-500"
          fill="none"
          id="icon"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 5a2 2 0 012-2h10a2 2 0 012 2v16l-7-3.5L5 21V5z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
